File: stack & queue/StackArray.java

public class StackArray {
    int[] array;
    int capacity;
    int top;

    StackArray(int capacity){
        this.capacity = capacity;
        top = -1;
        array = new int[capacity];
    }

    boolean isFull(){
        return top == capacity-1;
    }
    
    boolean isEmpty(){
        return top == -1;
    }

    void push(int key){
        if (isFull()){
            System.out.println("stack overflow");
            return;
        }
        array[++top] = key;
    }

    int pop(){
        int value;
        if (isEmpty())
            return -1;
        value = array[top];
        --top;
        return value;
    }

    int peek(){
        return array[top];
    }

    void print(){
        System.out.println();
        for (int i=0; i<top+1; i++)
            System.out.println(array[i]);
    }

    public static void main(String[] args) {
        
    }
}

File: stack/StackArray.js

class StackArray {
    constructor(capacity) {
        this.capacity = capacity;
        this.top = -1;
        this.items = [];
    }

    isFull() {
        return this.top === this.capacity-1;
    }

    isEmpty() {
        return this.top === -1;
    }

    peek() {
        if (this.top === -1) return -1;
        return this.items[this.top];
    }

    push(key) {
        if (this.isFull()) {
            console.log('stack overflow!!')
            return;
        }
        this.items[++this.top] = key;
    }

    pop() {
        if (this.isEmpty()) {
            return -1;
        }
        return this.items[this.top--];
    }

    clear() {
        this.items = [];
    }

    size() {
        return this.items.length;
    }

}

export default StackArray;

/*  Usage
    const stack = new StackArray(4);
    stack.push(1);
    stack.push(2);
    stack.push(3);
    stack.push(4);
    console.log(stack.pop())
    console.log(stack.peek())
*/